src/Types/ContextFactory.php

```diff
@@ -70,6 +70,26 @@ public function createForNamespace($namespace, $fileContents)
                 case T_NAMESPACE:
                     $currentNamespace = $this->parseNamespace($tokens);
                     break;
+                case T_CLASS:
+                    // Fast-forward the iterator through the class so that any
+                    // T_USE tokens found within are skipped - these are not
+                    // valid namespace use statements so should be ignored.
+                    $braceLevel = 0;
+                    $firstBraceFound = false;
+                    while ($tokens->valid() && ($braceLevel > 0 || !$firstBraceFound)) {
+                        if ($tokens->current() === '{') {
```

Comment:
Perhaps it is more reliable to ignore `use` statements after a class, interface or trait keyword is encountered and re-start checking for use statements after the namespace keyword is encountered again; this would prevent having to deal with nesting levels